sdk/qrcode/android/java/com/google/cardboard/sdk/qrcode/CardboardParamsUtils.java

```diff
@@ -71,14 +74,96 @@
     EXTERNAL_STORAGE
   };
 
+  /** Holds status for conversion from a URI to Cardboard device params. */
+  public static class UriToParamsStatus {
+    public static final int STATUS_OK = 0;
+    public static final int STATUS_UNEXPECTED_FORMAT = 1;
+    public static final int STATUS_CONNECTION_ERROR = 2;
+
+    public final int statusCode;
+    /** Only not null when statusCode is STATUS_OK. */
+    @Nullable
+    public final byte[] params;
+
+    public static UriToParamsStatus success(byte[] params) {
+      return new UriToParamsStatus(STATUS_OK, params);
+    }
+
+    public static UriToParamsStatus error(int statusCode) {
+      return new UriToParamsStatus(statusCode, null);
+    }
+
+    private UriToParamsStatus(int statusCode, @Nullable byte[] params) {
+      this.statusCode = statusCode;
+      this.params = params;
+    }
+  }
+
   /** URI of original cardboard QR code. */
   // TODO(b/170471141): Wifi connection shouldn't be required for Cardboard Viewer v1.0.
   private static final Uri URI_ORIGINAL_CARDBOARD_QR_CODE =
-      new Uri.Builder()
-          .scheme(HTTPS_SCHEME)
-          .authority(URI_HOST_GOOGLE_SHORT)
-          .appendEncodedPath(URI_PATH_CARDBOARD_HOME)
-          .build();
+          new Uri.Builder()
+                  .scheme(HTTPS_SCHEME)
+                  .authority(URI_HOST_GOOGLE_SHORT)
+                  .appendEncodedPath(URI_PATH_CARDBOARD_HOME)
+                  .build();
+
+  /**
+   * Converts a bytes array with an URI string to a Uri object.
+   * Obtains the physical parameters of a Cardboard headset from the Uri (as bytes) by
+   * calling {@code createFromUri}.
```

Comment:
I think this is outdated now, could be? (I don't see the call to `createFromUri` now)